folder-based hair and ACS sprites

ThePotatoGuy opened the pull request:
#8596 

# Key Changes
* hair and ACS sprite objects now support folder-based organization, basically similar to clothes:
    * `h/<hair name>/front.png` or `h/<hair name>/0.png`
    * `a/<acs name>/0.png`
* sprite JSONs support the new format as well
   * this is based on a version - so version 4 and beyond must be folder-based
   * ver 3 sprites will continue to be supported but a warning will be logged to consider updating the version
* a simple bash script for renaming using git is provided, but a full-fledged spritepack updater will be added in #9729 

# Testing
* verify that all current sprites load correctly
* verify spritepacks with jsons **do not break** - testing of the new json props should be in #9729 for ease of testing

Booplicate reviewed (commented):
Maybe instead of adding another param (`use_folders`), use spritepack version to determine how it should be loaded? More params > harder for people to understand our system and what params to use. We should also discourage people from using the old structure, having an upgrade flag instead of an option is better for that case imo.

We should also update our wiki

ThePotatoGuy commented:
changed to use version to determine `use_folders`.

Also rebased commits so all main code changes are in 1 (next time I'll do a better job of splitting these out)

Booplicate commented:
Besides `use_numeric_layer_ids` not using `version` I think it's great

ThePotatoGuy commented:
down the line that might be an option, the key benefit of numeric ids is more splits beyond 3, it's just not necessary now so.


i'll merge tonight
